### PR TITLE
Fix oauth2 consent page ambiguity

### PR DESCRIPTION
**Pull request by brionmario**
## Purpose
Fixing ambiguity in OAuth consent page "Approve once" and "Approve Always" words.

## Goals
Add two buttons, **Approve** and **Deny** & a checkbox for remember approval.
- **Deny** button covers Deny approval
- **Allow** button covers approve current session only
- Checking the "**Remember my approval**" checkbox covers  approve all sessions

Fixes https://github.com/wso2/product-is/issues/8588

## Approach
*OIDC Consent Page*
Screen Shot 2020-11-12 at 1 11 43 AM

*OAuth2 Consent Page*
Screen Shot 2020-11-12 at 1 11 33 AM

### CODE REVIEW

**Review by thivi — APPROVED** [review]

**Comment by thivi** [comment]
I think it will be better if we can hide the "MyAccount would need to:" text (in the second image) if there are no attributes to be approved. WDYT?

**Comment by jeradrutnam** [comment]
Related: https://github.com/wso2/product-is/issues/10478

File: apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources.properties [hunk 1]
```diff
@@ -33,13 +33,15 @@ account.resend.email.success=Email sent successfully.
 account.resend.email.fail=Email sent fail.
 user.tenant.domain.mismatch.message=Application you are trying to access does not allow users from your organization.
 remember.me=Remember me on this computer
+remember.my.approval=Remember my approval
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources.properties:36]:
```suggestion
remember.my.consent=Remember my consent
```

File: apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_fr_FR.properties [hunk 2]
```diff
@@ -32,6 +32,7 @@ password.reset.pending=La réinitialisation du mot de passe est requise. Un lien
 account.resend.email.success=L'e-mail a été envoyé avec succès.
 account.resend.email.fail=L'envoi de l'email a échoué.
 user.tenant.domain.mismatch.message=L'application à laquelle vous essayez d'accéder n'autorise pas les utilisateurs de votre organisation.
+remember.my.approval=Souviens-toi de mon approbation
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_fr_FR.properties:35]:
```suggestion
remember.my.consent=Souviens-toi de mon consentement
```

File: apps/authentication-portal/src/main/webapp/oauth2_authz.jsp [hunk 3]
```diff
@@ -145,26 +145,21 @@
...
+                        <div class="field">
+                            <div class="ui checkbox">
+                                <input
+                                    tabindex="3"
+                                    type="checkbox"
+                                    id="rememberApproval"
+                                    name="rememberApproval"
+                                    data-testid="oauth2-consent-page-remember-approval-checkbox"
+                                />
+                                <label><%=AuthenticationEndpointUtil.i18n(resourceBundle, "remember.my.approval")%></label>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/oauth2_authz.jsp:158]:
```suggestion
                                <label><%=AuthenticationEndpointUtil.i18n(resourceBundle, "remember.my.consent")%></label>
```

File: apps/authentication-portal/src/main/webapp/oauth2_consent.jsp [hunk 4]
```diff
@@ -251,26 +251,19 @@
...
+                                <div class="field">
+                                    <div class="ui checkbox">
+                                        <input
+                                            tabindex="3"
+                                            type="checkbox"
+                                            id="rememberApproval"
+                                            name="rememberApproval"
+                                            data-testid="consent-page-remember-approval-checkbox"
+                                        />
+                                        <label><%=AuthenticationEndpointUtil.i18n(resourceBundle, "remember.my.approval")%></label>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/oauth2_consent.jsp:264]:
```suggestion
                                        <label><%=AuthenticationEndpointUtil.i18n(resourceBundle, "remember.my.consent")%></label>
```

**Comment by jeradrutnam** [comment]
> I think it will be better if we can hide the "MyAccount would need to:" text (in the second image) if there are no attributes to be approved. WDYT?

We need remove this page if there is no consent required. I think that how it works currently.
And above is just a screenshot from the UI. Not an actual populated screen.

**Pull request merged**